keepsorted/keep_sorted_test.go

```diff
@@ -717,20 +722,55 @@ i
 				{3, startDirective},
 			},
 		},
+		{
+			name: "BadOption",
+			in: `
+// keep-sorted-test start foo=bar block=yes
+0
+1
+2
+// keep-sorted-test end
+`,
+
+			wantBlocks: []block{
+				{
+					metadata: defaultMetadataWith(func() blockOptions {
+						var opts blockOptions
+						opts.Block = true
+						opts.setCommentMarker("//")
+						return opts
+					}()),
+					start: 1,
+					end:   5,
+					lines: []string{"0", "1", "2"},
+				},
+			},
+			wantWarnings: map[int]int{
+				1: 1,
+			},
+		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
 			initZerolog(t)
 			if tc.include == nil {
 				tc.include = func(start, end int) bool { return true }
 			}
 
-			gotBlocks, gotIncompleteBlocks := New("keep-sorted-test", BlockOptions{}).newBlocks(strings.Split(tc.in, "\n"), 0, tc.include)
+			gotBlocks, gotIncompleteBlocks, gotWarnings := New("keep-sorted-test", BlockOptions{}).newBlocks(strings.Split(tc.in, "\n"), 0, tc.include)
 			if diff := cmp.Diff(tc.wantBlocks, gotBlocks, cmp.AllowUnexported(block{}, blockMetadata{}, blockOptions{})); diff != "" {
 				t.Errorf("blocks diff (-want +got):\n%s", diff)
 			}
 			if diff := cmp.Diff(tc.wantIncompleteBlocks, gotIncompleteBlocks, cmp.AllowUnexported(incompleteBlock{})); diff != "" {
 				t.Errorf("incompleteBlocks diff (-want +got):\n%s", diff)
 			}
+
+			warnings := make(map[int]int)
+			for k, v := range gotWarnings {
+				warnings[k] = len(v)
```

Comment:
Is there any reason we use `len(v)` instead of, say, the string equivalent of the warning (which is what ends up in the findings)? That way we can make sure that the actual warning text is what we expect. There only seems to be one warning in this whole test, so it doesn't seem like it'll be too verbose or complex or the likes.